clientv3/watch.go

```diff
@@ -484,9 +490,35 @@ func (w *watchGrpcStream) run() {
 				closeErr = err
 				return
 			}
-			if wc, closeErr = w.newWatchClient(); closeErr != nil {
+
+			clientOpenC := make(chan openWatchClientResponse, 1)
+			go func() {
+				owc, err := w.openWatchClient(w.ctx)
+				clientOpenC <- openWatchClientResponse{wc: owc, err: err}
+			}()
+
+			for {
+				select {
+				case opc := <-clientOpenC:
```

Comment:
@yudai OK. Now I understand this better. We move part of the logic into serveWatchClient to solve race conditions?